src/passes/pass.cpp

```diff
@@ -372,144 +372,128 @@ void PassRegistry::registerPasses() {
   //   "lower-i64", "lowers i64 into pairs of i32s", createLowerInt64Pass);
 }
 
+void PassRunner::addIfDWARFAllowed(std::string passName) {
+  auto pass = PassRegistry::get()->createPass(passName);
+  if (!pass->invalidatesDWARF() ||
+      !Debug::shouldPreserveDWARF(options, *wasm)) {
+    doAdd(std::move(pass));
```

Comment:
When the pass does not support DWARF, In which case we don't add the pass and in which case we just warn? `doAdd` seems to just warn, but this function, which calls `doAdd`, seems not call `doAdd` in the first place.

Reply:
`doAdd` always adds. It may warn as needed.

This method `addIfDWARFAllowed` should be something like `addPassIfNoDWARFWarningWouldBeShown`. That is, this method is called when the addition is optional, and in that case it's best to not add the pass if there are issues.